helix-core/src/test/java/org/apache/helix/manager/zk/TestZkBaseDataAccessor.java

```diff
@@ -188,6 +212,72 @@ public void testSyncCreate() {
     System.out.println("END " + testName + " at " + new Date(System.currentTimeMillis()));
   }
 
+  @Test
+  public void testDefaultAccessorCreateCustomData() {
+    String className = TestHelper.getTestClassName();
+    String methodName = TestHelper.getTestMethodName();
+    String testName = className + "_" + methodName;
+    System.out.println("START " + testName + " at " + new Date(System.currentTimeMillis()));
+
+    String path = String.format("/%s/%s", _rootPath, "msg_0");
+
+    ZkBaseDataAccessor defaultAccessor = new ZkBaseDataAccessor(ZK_ADDR);
+
+    List<Integer> l0 = ImmutableList.of(1, 2, 3);
+    boolean createResult = defaultAccessor.create(path, l0, AccessOption.PERSISTENT);
+    // The result is expected to be false because the list is not ZNRecord
+    Assert.assertFalse(createResult);
+    createResult = defaultAccessor.create(path, new ZNRecord("test"), AccessOption.PERSISTENT);
+    // The result is expected to be true
+    Assert.assertTrue(createResult);
+  }
+
+  @Test
+  public void testCustomAccessorCreateZnRecord() {
+    String className = TestHelper.getTestClassName();
+    String methodName = TestHelper.getTestMethodName();
+    String testName = className + "_" + methodName;
+    System.out.println("START " + testName + " at " + new Date(System.currentTimeMillis()));
+
+    String path = String.format("/%s/%s", _rootPath, "msg_0");
+
+    ZkBaseDataAccessor customDataAccessor = new ZkBaseDataAccessor(ZK_ADDR, LIST_SERIALIZER);
```

Comment:
Please close this accessor at the end of the test to prevent a ZkClient leak.

Reply:
done.